packages/react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -597,8 +597,10 @@ const DetailsListInner: React.ComponentType<IDetailsListInnerProps> = (
     onBlur: focusZoneProps && focusZoneProps.onBlur ? focusZoneProps.onBlur : onBlur,
   };
 
+  const FinalGroupedList = groups && groupProps?.groupedListAs ? groupProps.groupedListAs : GroupedList;
```

Comment:
If someone uses the new GroupedList within DetailsList, it looks like it won't handle using the correct semantics for being used within a grid, instead of a treegrid.

I think it'd make sense to hold off on the DetailsList change until GroupedListV2 actually supports being used within DetailsList.

Reply:
What changes do we need to make to `GroupedListV2` to make the semantics correct?

Reply:
Made a follow up issue to address this: #24796